doc/news/changes/major/20240909Bangerth

```diff
@@ -1,6 +1,9 @@
-Changed: The SmartPointer class has been renamed to ObserverPointer,
+Changed: The `SmartPointer` class has been renamed to ObserverPointer,
```

Comment:
I'm marking up the old class name because doxygen will at some point no longer recognize it as a name.